ironic/drivers/modules/drac/raid.py

```diff
@@ -996,6 +1064,44 @@ def _create_virtual_disks(task, node):
     return _commit_to_controllers(node, controllers)
 
 
+def controller_supports_ehba_mode(settings, controller_fqdd):
```

Comment:
This is detecting if the RAID controller is in HBA mode.  I'm thinking that you want to check if the controller supports HBA mode.
I checked a storage node that has BOSS+HBA, and in that config, RAIDCurrentControllerMode does not exist.
I'm thinking that this method should try to get the RAIDCurrentControllerMode.  If it doesn't exist, then the controller doesn't support HBA mode.  Next, if it was returned, then check possible_values and see if that contains "HBA".  If not, then the controller doesn't support HBA mode, otherwise it does.

Reply:
Got your point. I will address this in next commit

Reply:
Done